lib/config.rb

```diff
@@ -48,6 +49,14 @@ def self.load_files(*sources)
 
     config.add_source!(Sources::EnvSource.new(ENV)) if Config.use_env
 
+    if defined?(::Rails::Railtie) && Config.use_rails_credentials
+      if Rails.version < '7.1'
+        config.add_source!(Sources::HashSource.new(secret: Rails.application.secrets.to_h.deep_stringify_keys))
+      else
+        config.add_source!(Sources::HashSource.new(secret: Rails.application.credentials.config))
```

Comment:
I think `Rails.application.credentials.config` returns a hash with symbol keys. I think for merging to work correctly, we need to `.deep_stringify_keys` here.